src/main/java/org/opensearch/neuralsearch/search/query/TopDocsMerger.java

```diff
@@ -24,8 +24,7 @@
 @RequiredArgsConstructor(access = AccessLevel.PACKAGE)
 class TopDocsMerger {
 
-    private HybridQueryScoreDocsMerger<ScoreDoc> scoreDocsMerger;
-    private HybridQueryScoreDocsMerger<FieldDoc> fieldDocsMerger;
+    private HybridQueryScoreDocsMerger scoreDocsMerger;
```

Comment:
```suggestion
    private HybridQueryScoreDocsMerger docsMerger;
```
use more generic name, now it's one merger for both scoreDocs and fieldDocs